src/cli/util.ts

```diff
@@ -0,0 +1,59 @@
+import {getCliFields, IConfigurationField, IConfigurationModule, isConfigurationModule} from "../util/config";
+import {Command} from "commander";
+
+export function generateCommanderOptions(command: Command, optionDescription: IConfigurationModule): void {
+  const cliOptions = getCliFields(optionDescription);
+  cliOptions.forEach((cliOption) => {
+    let flag = "";
+    if(cliOption.cli.short) {
+      flag += `-${cliOption.cli.short}, `;
+    }
+    flag += `--${cliOption.cli.flag}`;
+    if(cliOption.type != Boolean) {
+      flag = flag + ` <${cliOption.name}>`;
+    }
+    const defaultProcess = (arg) => {
+      return arg;
+    };
+    command.option(flag, cliOption.description, cliOption.process || defaultProcess);
+
+  });
+}
+
+/**
+ * This is awful,
+ * @param options
+ * @param optionDescription
+ */
+export function optionsToConfig<T>(options: {[key: string]: string}, optionDescription: IConfigurationModule): Partial<T> {
+  const config = {};
+  optionDescription.fields.forEach((field) => {
+    if (isConfigurationModule(field)) {
+      processModule(options, field, config);
+    } else {
+      processField(field, options, config);
+    }
+  });
+  return config;
+}
+
+function processModule(options: { [p: string]: string }, field, config) {
+  const childConfig = optionsToConfig(options, field as IConfigurationModule);
+  if (Object.keys(childConfig).length > 0) {
```

Comment:
Why is this check needed here?

Reply:
If you pass empty object to toml library it produces weird invalid toml